### ISSUE TITLE
Issue with CORS

### ISSUE THREAD

**Issue by boganiLuca** [shamblett/wilt#31]
I am trying to develop a Flutter Web application using Wilt, but I am still trying to figure it out if it's working with the latest CouchDB version, or there's something else that I'm missing here.
The CouchDB installation is pretty simple: version 3.1.1, no HTTPS, just a database with a single document. CORS enabled, as shown in the screenshot, but, again, nothing too convoluted:
![image](https://user-images.githubusercontent.com/44212022/129028332-fb4baf22-0854-4c6d-80f0-03a6674d120a.png)

I wrote a basic a flutter app that tries to retrieve all documents from the database: it's just a few lines of code different from the example, actually.

Main:
```
void main() {
  const hostName = 'localhost';
  const serverPort = 5984;
  const useSSL = false;

  /// Create a test client
  final wilting = Wilt(hostName, port: serverPort, useSSL: useSSL);
  runApp(MyApp(wilting));
}
```

MyApp:
```
class MyHomePage extends StatefulWidget {
  MyHomePage({required this.title, required this.database}) : super();
  final String title;
  final Wilt database;
  @override
  _MyHomePageState createState() => _MyHomePageState(database);
}

class _MyHomePageState extends State<MyHomePage> {
  final Wilt db;

  _MyHomePageState(this.db);

  @override
  Widget build(BuildContext context) {
    db.db = 'prova';   //<- database name

    db.login(userName, userPassword);
    return Scaffold(
        body: FutureBuilder(
            future: db.getAllDocs(),
            builder: (context, snapshot) {
              if (!snapshot.hasData && snapshot.connectionState != ConnectionState.done) {
                return Center(child: CircularProgressIndicator());
              }

              JsonObjectLite<dynamic> response = snapshot.data as JsonObjectLite<dynamic>;
              List<Widget> children = [];
              response.forEach((key, value) {
                print("$key: $value");
              });

              if (children.isEmpty) {
                return Center(child: Text("Nessun dato da mostrare"));
              }
              return Center(
                child: Column(),
              );
            }));
 }
```

When I try running this on Chrome or Edge directly from Android Studio, I get a CORS error:
![image](https://user-images.githubusercontent.com/44212022/129030938-129497f6-1d11-40b6-abca-6b17fbeb38c1.png)

I thought that this might be caused by running from the IDE, so I tried setting up a local webserver to do some test, but the few times I managed to get it up and running, I always get the same CORS error (not that I managed it so many times, I'm also having difficulties finding a simple webserver for Windows that would run Flutter Web apps). 
The latest tool I'm using is `webdev` for dart, and it still shows me the same error as in the screenshot above. In the meantime, I noticed that by clicking the link `http://localhost:5984/prova/_all_docs` shown in the browser console, the page opens just fine and shows me the JSON for the documents in the database. 

The only time the error isn't there is when I run Chrome with the argument `--disable-web-security`: while it may be enough for development purpose, I won't be able to use that argument when the app gets released, so it's not really a solution, it would only postpone the problem.

So, to sum it up, I'm resorting to ask you if you could just test if there's any problem with the setup I described above, because I'm really running out of ideas on how to solve that CORS issue.

**Comment by boganiLuca** [comment]
After some more digging, I discovered where the problem is: the TCP port is relevant for CORS, and omitting it from allowed domains implies that only the default port is allowed (80 for http, 443 for https). I thought that omitting it meant "any port allowed", but that is not the case. 
Anyway, I tested it and by allowing the full URL in CouchDB (i.e. in the screenshot 'http://localhost:58316'), Wilt works just fine. 
Closing this.
